test/ig/IGExporter.index.test.ts

```diff
@@ -0,0 +1,272 @@
+import fs from 'fs-extra';
+import path from 'path';
+import temp from 'temp';
+import { IGExporter } from '../../src/ig';
+import { Package } from '../../src/export';
+import { loggerSpy } from '../testhelpers/loggerSpy';
+import { minimalConfig } from './fixtures/minimalConfig';
+
+describe('IGExporter', () => {
+  // Track temp files/folders for cleanup
+  temp.track();
+
+  describe('#index', () => {
+    let tempOut: string;
+    let indexMarkdownPageContentPath: string;
+    let indexXMLPageContentPath: string;
+    let indexMarkdownPagesPath: string;
+    let indexXMLPagesPath: string;
+    let igPath: string;
+
+    beforeAll(() => {
+      tempOut = temp.mkdirSync('sushi-test');
+      indexMarkdownPageContentPath = path.join(tempOut, 'input', 'pagecontent', 'index.md');
+      indexXMLPageContentPath = path.join(tempOut, 'input', 'pagecontent', 'index.xml');
+      indexMarkdownPagesPath = path.join(tempOut, 'input', 'pages', 'index.md');
+      indexXMLPagesPath = path.join(tempOut, 'input', 'pages', 'index.xml');
+      igPath = path.join(tempOut, 'input', 'ImplementationGuide-fhir.us.minimal.json');
+    });
+
+    afterAll(() => {
+      temp.cleanupSync();
+    });
+
+    beforeEach(() => {
+      loggerSpy.reset();
+    });
+
+    it('should not export index file when config.indexPageContent is not defined and none provided', () => {
+      const pkg = new Package(null, minimalConfig);
+      const exporter = new IGExporter(pkg, null, '');
+      exporter.initIG();
+      exporter.addIndex(tempOut);
+      exporter.addImplementationGuide(tempOut);
+      expect(fs.existsSync(indexMarkdownPageContentPath)).toBeFalsy();
+      expect(fs.existsSync(indexXMLPageContentPath)).toBeFalsy();
+      expect(fs.existsSync(indexMarkdownPagesPath)).toBeFalsy();
+      expect(fs.existsSync(indexXMLPagesPath)).toBeFalsy();
+
+      // Checks that the index.md file is not added to IG definition
+      expect(fs.existsSync(igPath)).toBeTruthy();
+      const igContent = fs.readJSONSync(igPath);
+      expect(igContent.definition.page.page).not.toContainEqual({
+        nameUrl: 'index.html',
+        title: 'Home',
+        generation: 'markdown'
+      });
+
+      // No warnings or errors logged
+      expect(loggerSpy.getAllMessages()).toHaveLength(loggerSpy.getAllMessages('info').length);
```

Comment:
I'm going to have to remember this one for later.

Reply:
Or we could add a method to loggerSpy: `isClean()` (or something like that).